apm-agent-core/src/main/java/co/elastic/apm/agent/report/ApmServerClient.java

```diff
@@ -234,14 +233,31 @@ void onConnectionError() {
             } catch (Exception e) {
                 logger.debug("Exception while interacting with APM Server", e);
             } finally {
-                if (connection != null) {
-                    connection.disconnect();
-                }
+                closeInputStream(connection);
             }
         }
         return results;
     }
 
+    private void closeInputStream(@Nullable HttpURLConnection connection) {
```

Comment:
also close the stream in `co.elastic.apm.agent.report.IntakeV2ReportingEventHandler`

Reply:
added to HttpUtils as static method